### DISCUSSION TITLE
Write your json format as a reply

### DISCUSSION THREAD

**Discussion by diegoq48** [uprm-inso4101-2023-2024-s2/semester-project-personalfinancemanager#43] [General]
Write your json format as a reply

**Comment by NathBat09** [comment]
For render my pie chart I need:

Categories and their values. I use this format for the sample data:
[{ category: 'Food', value: 300 },
{ category: 'Housing', value: 500 },
{ category: 'Entertainment', value: 200 },
{ category: 'Others', value: 100 }]

I don't know how to placed it in a JSON format that works for the other components too, but in summary, I need:
The **categories** and the **total value of all the expenses in that category**.

And maybe we can do a pie chart for the incomes like: salary, investments, etc. So I would need in that case:
The **categories** and the **total value of all the incomes in that category**.

Maybe something like this could work for that specific component: 

{
  "data": [
    { "category": "Food", "value": 300 },
    { "category": "Housing", "value": 500 },
    { "category": "Entertainment", "value": 200 },
    { "category": "Others", "value": 100 }
  ]

**Comment by fernandopizarro1** [comment]
### BarChart
---
The data needed for the barchart is an **array of numeric values, in this case income since they are all positive, and the name of each of those values:**
`const [data] = useState([
    { value: 300, name: "Yaira" },
    { value: 125, name: "Samantha" },
    { value: 200, name: "Lisa" },
    { value: 250, name: "Angelica" },
    { value: 100, name: "Mathew" },
    { value: 75, name: "Luis" },
    { value: 25, name: "Fernando" },
    { value: 215, name: "Angel" },
    { value: 165, name: "Rosa" },
    { value: 201, name: "Michael" },
    { value: 34, name: "John" }
  ]);`
The "name" part can be changed to category if it makes the process of implementing them easier for generalization.
### Diverging Bar Chart 
---
The data needed for this barchart is also an array consisting of the name of the categories and the total value of these categories: 
`const [data] = useState([
        { value: 300, category: "Jobs" },
        { value: -125, category: "Bank" },
        { value: -200, category: "Car" },
        { value: 250, category: "Properties" },
        { value: 100, category: "Stocks" },
        { value: -75, category: "Jewelry" },
        { value: 25, category: "Betting" },
        { value: -180, category: "Food" },
        { value: -80, category: "House" },
        { value: 201, category: "Bonds" },
        { value: 34, category: "Investments" }
      ]);`
### HOWEVER, this chart has both income and expenses, and if you take a close look into the code provided you can see that income is represented as a **_positive_** numeric value and the expenses are represented as **_negative_** numeric values. Please make sure to distinguish these.

**Comment by AngelMartinez33** [comment]
The data needed for the line chart:
[INCOME] and [EXPENSES] per month.

The chart displays a line for expense data (total expenses per month) and another line for income data (total income per month). Categories are not needed in this graph.

"incomeData" : [
{"month": January, "value": 500},
{"month": February, "value": 475},
{"month": March, "value": 610}
]

"expensesData" : [
{"month": January, "value": 300},
{"month": February, "value": 234},
{"month": March, "value": 401}
]

**Comment by ActuallySeb** [comment]
I recommend that the JSON contain an object that would contain the user information; an array of the expenses containing the year, month, type of expense and amount; and a different array containing year, month, and income generated.

**Comment by diegoq48** [comment]
What do you guys think of the following "{
  "userInfo": {
    "name": "User Name",
    "id": "UserID",
    "dateJoined": "YYYY-MM-DD"
  },
  "financialData": {
    "monthlyDetails": [
      {
        "month": "January",
        "year": 2024,
        "incomes": [
          { "category": "Salary", "value": 2000 },
          { "category": "Investments", "value": 500 }
        ],
        "expenses": [
          { "category": "Food", "value": 300 },
          { "category": "Housing", "value": 500 },
          { "category": "Entertainment", "value": 200 },
          { "category": "Others", "value": 100 }
        ]
      },
      {
        "month": "February",
        "year": 2024,
        // Additional months follow the same structure
      }
    ],
    "chartData": {
      "categories": [
        "Salary", "Investments", "Food", "Housing", "Entertainment", "Others"
        // Add more categories as needed
      ],
      "barChart": [
        // Example data for a specific chart type if needed
      ],
      "lineChart": {
        "income": [
          { "month": "January", "value": 2500 },
          { "month": "February", "value": 3000 }
          // Additional data points
        ],
        "expenses": [
          { "month": "January", "value": 1100 },
          { "month": "February", "value": 1200 }
          // Additional data points
        ]
      }
    }
  }
}
"

**Reply by fernandopizarro1** [reply]
I think that works great with all the data we have!
